### PR TITLE
Refactor library

### PR DESCRIPTION
**Pull request by luchaos**
- Consolidate functions of large library files into separate feature files. Reduces amount of lines of hard to maintain files,  groups related queries.
     - `database/game.php` 1155 lines
     - `database/achievement.php` 1203 lines
     - `database/user.php` 1838 lines
- Delete expired email verification tokens in daily cron
- Remove unused functions and parameters
- Remove superfluous dorequest comments
- Rename cookie functions

### CODE REVIEW

File: lib/database/achievement-creator.php [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+{
+    sanitize_sql_inputs($user);
+
+    $retVal = [];
+    $query = "SELECT COUNT(a.GameID) AS AchievementCount, c.Name AS ConsoleName
+              FROM Achievements as a
+              LEFT JOIN GameData AS gd ON gd.ID = a.GameID
+              LEFT JOIN Console AS c ON c.ID = gd.ConsoleID
+              WHERE a.Author = '$user'
+              AND a.Flags = '3'
```
Comment (Jamiras) [inline lib/database/achievement-creator.php:16]: Opportunity to use `AchievementType::CORE`

Comment (Jamiras) [reply lib/database/achievement-creator.php:16]: Repeated several times in this file.

File: lib/database/player-achievement.php [hunk 2]
```diff
@@ -0,0 +1,610 @@
...
+{
+    sanitize_sql_inputs($user, $achIDToAward, $isHardcore);
+    settype($achIDToAward, 'integer');
+    settype($isHardcore, 'integer');
+
+    $retVal = [];
+    $retVal['Success'] = false;
+
+    if ($achIDToAward <= 0) {
+        $retVal['Error'] = "Achievement ID <= 0! Cannot unlocl";
```
Comment (Jamiras) [inline lib/database/player-achievement.php:44]: unlock misspelled.

File: lib/database/player-achievement.php [hunk 3]
```diff
@@ -0,0 +1,610 @@
...
+
+    // Update GameData
+    // Removed: this needs rethinking! //##SD TBD
+    // recalculateTrueRatio( $gameID );    // Heavy!
+    // Add TA to the player for this achievement, NOW that the TA value has been recalculated
+    // Select the NEW TA from this achievement, as it has just been recalculated
+    $query = "SELECT TrueRatio
+              FROM Achievements
+              WHERE ID='$achIDToAward'";
+    $dbResult = s_mysql_query($query);
```
Comment (Jamiras) [inline lib/database/player-achievement.php:143]: `TrueRatio` is returned as part of `GetAchievementMetadataJSON`. This query is just to get the updated value, which didn't occur on line 137.

File: lib/database/player-game.php [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+use RA\ActivityType;
+
+function testFullyCompletedGame($gameID, $user, $isHardcore, $postMastery): array
+{
+    sanitize_sql_inputs($gameID, $user, $isHardcore);
+    settype($isHardcore, 'integer');
+
+    $query = "SELECT COUNT(ach.ID) AS NumAch, COUNT(aw.AchievementID) AS NumAwarded FROM Achievements AS ach
+              LEFT JOIN Awarded AS aw ON aw.AchievementID = ach.ID AND aw.User = '$user' AND aw.HardcoreMode = $isHardcore 
+              WHERE ach.GameID = $gameID AND ach.Flags = 3 ";
```
Comment (Jamiras) [inline lib/database/player-game.php:12]: more opportunities for `AchievementType::CORE` in this file. probably should spin up a separate PR

**Review by Jamiras — COMMENTED** [review]
I gave up looking at the code about halfway through this. It seems mostly like it's just moving stuff around, but there's some renames. 

Just trying to use the site, pretty much every page complained about not being able to find `getUserRank`. It looks like you forgot `player-rank.php` in `functions.php`.

After fixing that, I noticed that the user page wasn't fully rendering. A view source reports:
```<b>Fatal error</b>:  Uncaught Error: Call to undefined function HasPatreonBadge() in /home/brian/source/RAWeb/public/userInfo.php:389```

I also found this error on the "view latest hashes" page:
```Uncaught TypeError: getHashList(): Argument #3 ($searchedHash) must be of type string, null given, called in /home/brian/source/RAWeb/public/latesthasheslinked.php on line 23 and defined in /home/brian/source/RAWeb/lib/database/hash.php:77```

File: lib/database/player-game.php [hunk 5]
```diff
@@ -312,7 +314,7 @@ function getGameRecentPlayers($gameID, $maximum_results = 0): array
 
     $query = "SELECT ua.ID as UserID, ua.User, ua.RichPresenceMsgDate AS Date, ua.RichPresenceMsg AS Activity
               FROM UserAccounts AS ua
-              WHERE ua.LastGameID = $gameID AND ua.Permissions >= 0
+              WHERE ua.LastGameID = $gameID AND ua.Permissions >= " . Permissions::Registered . "
```
Comment (Jamiras) [inline lib/database/player-game.php:317]: 0 is `Permissions::Unregistered`. Was this change intentional?

Comment (luchaos) [reply lib/database/player-game.php:317]: Oh, no 🙈  Thanks!

File: lib/database/player-achievement.php (hunk 3 above)
Comment (luchaos) [reply lib/database/player-achievement.php:143]: Hrm, true, that has to be a left-over. I guess it would be ok to remove all of the TruePoints related code as it couldn't have had any impact for quite some time?

Comment (Jamiras) [reply lib/database/player-achievement.php:143]: Not all. Just up through line 150. The next query will update the white points for the user, just using the current TrueRatio value.

Comment (luchaos) [reply lib/database/player-achievement.php]: Makes sense, changed it accordingly

**Review by Jamiras — APPROVED** [review]
Don't see any other obvious errors when doing a spot-check.

**Comment by luchaos** [comment]
Awesome, thanks! I went through pretty much all the pages another time, too, to see if anything else broke.
